### PR TITLE
github: run nested suite when commit is pushed to release branch

### PR DESCRIPTION
**Pull request by bboozzoo**
A cherry-picks to the relase branches are sometimes applied by directly pushing
the repository rather than via a pull request. Make sure that nested tests are
run when pushing to the release branch.

### CODE REVIEW

**Review by stolowski — APPROVED** [review]
LGTM

**Review by mvo5 — APPROVED** [review]
thanks

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -286,7 +286,9 @@ jobs:
               echo "::set-output name=already-ran::true"
           fi
     - name: Run spread tests
-      if: "contains(github.event.pull_request.labels.*.name, 'Run nested') && steps.cached-results.outputs.already-ran != 'true'"
+      # run if the commit is pushed to the release/* branch or there is a 'Run
+      # nested' label set on the PR
+      if: "(contains(github.event.pull_request.labels.*.name, 'Run nested') || contains(github.ref, 'refs/heads/release/')) && steps.cached-results.outputs.already-ran != 'true'"
```
Comment (mvo5) [inline .github/workflows/test.yaml:291]: Having this is great. But I think the current spread has some issue when it tries to run under release/**, e.g.:
https://github.com/snapcore/snapd/runs/1224993441

Comment (anonymouse64) [reply .github/workflows/test.yaml:291]: the run you linked to failed on unit tests, not on running spread actually, did you mean a different run?

**Review by anonymouse64 — APPROVED** [review]

**Pull request merged**